ci: add ors-config to release artifacts

jhaeu opened the pull request:
### Pull Request Checklist
<!--- Please make sure you have completed the following items BEFORE submitting a pull request (put an x in each box
when you have checked you have done them): -->
- [ ] 1. I have [**rebased**][rebase] the latest version of the main branch into my feature branch and all conflicts
         have been resolved.
- [ ] 2. I have added information about the change/addition to functionality to the CHANGELOG.md file under the
         [Unreleased] heading.
- [ ] 3. I have documented my code using JDocs tags.
- [ ] 4. I have removed unnecessary commented out code, imports and System.out.println statements.
- [ ] 5. I have written JUnit tests for any new methods/classes and ensured that they pass.
- [ ] 6. I have created API tests for any new functionality exposed to the API.
- [ ] 7. If changes/additions are made to the ors-config.json file, I have added these to the [ors config documentation][config]
         along with a short description of what it is for, and documented this in the Pull Request (below).
- [ ] 8. I have built graphs with my code of the Heidelberg.osm.gz file and run the api-tests with all test passing
- [ ] 9. I have referenced the Issue Number in the Pull Request (if the changes were from an issue).
- [ ] 10. For new features or changes involving building of graphs, I have tested on a larger dataset
          (at least Germany), and the graphs build without problems (i.e. no out-of-memory errors).
- [ ] 11. For new features or changes involving the graphbuilding process (i.e. changing encoders, updating the
          importer etc.), I have generated longer distance routes for the affected profiles with different options
          (avoid features, max weight etc.) and compared these with the routes of the same parameters and start/end
          points generated from the current live ORS.
          If there are differences then the reasoning for these **MUST** be documented in the pull request.
- [ ] 12. I have written in the Pull Request information about the changes made including their intended usage
          and why the change was needed.
- [ ] 13. For changes touching the API documentation, I have tested that the API playground [renders correctly][api].

Fixes # .

### Information about the changes
- Key functionality added:
- Reason for change:

### Examples and reasons for differences between live ORS routes, and those generated from this pull request
-

### Required changes to ors config (if applicable)
-

[config]: https://GIScience.github.io/openrouteservice/run-instance/configuration/
[api]: https://gitlab.gistools.geog.uni-heidelberg.de/giscience/openrouteservice-infrastructure/ors-docs-api#test-new-ors-documentation
[rebase]: https://github.com/GIScience/openrouteservice/blob/main/CONTRIBUTE.md#pull-request-guidelines

jhaeu committed:
ci: add ors-config to release artifacts

MichaelsJP commented on docs/run-instance/configuration/ors/engine/profiles.md:
```diff
@@ -58,6 +58,7 @@ Properties beneath `ors.engine.profiles.*.encoder_options`:
 | turn_costs               | boolean | Should turn restrictions be respected.                                                                                                                                           | `turn_costs=true`              |
 | problematic_speed_factor | number  | For wheelchair profile only! Travel speeds on edges classified as problematic for wheelchair users are multiplied by this factor, use to set slow traveling speeds on such ways  | `problematic_speed_factor=0.7` |
 | preferred_speed_factor   | number  | For wheelchair profile only! Travel speeds on edges classified as preferable for wheelchair users are multiplied by this factor, use to set faster traveling speeds on such ways | `preferred_speed_factor=1.2`   |
+use_acceleration
```
```suggestion
```

MichaelsJP commented on .github/workflows/release-artifacts.yml:
```diff
@@ -52,3 +52,23 @@ jobs:
+          upload_url: ${{ github.event.release.upload_url }}
+          asset_path: ./ors-config.env
+          asset_name: ors-config.env
+          asset_content_type: text/plain
```
| actions/upload-release-asset
This action is archived and unmaintained. They recommend the following action: https://github.com/softprops/action-gh-release

Maybe worth updating the other upload as well.

jhaeu committed:
chore: removed accidentally added word from docu

Co-authored-by: Julian Psotta <22392104+MichaelsJP@users.noreply.github.com>

MichaelsJP committed:
feat(workflow): Optimize the artifacts workflow

This combines the artifact upload into one action call. The maven cache is optimized as well.

MichaelsJP committed:
chore(workflow): Update all actions/checkout to version 4

MichaelsJP committed:
fix(workflow): Fix wrong indentation

MichaelsJP committed:
fix(workflow): Fix missing cache dependency path

MichaelsJP committed:
fix(workflow): Fix the wrong step order

MichaelsJP reviewed (approved):
LGTM. Tested in a fork PR
